app/src/ui/toolbar/push-pull-button.tsx

```diff
@@ -273,51 +153,162 @@ const forcePushIcon: OcticonSymbol.OcticonSymbolType = {
+  }
+
+  private closeDropdown() {
+    this.props.dispatcher.closeFoldout(FoldoutType.PushPull)
+  }
+
   private push = () => {
+    this.closeDropdown()
     this.props.dispatcher.push(this.props.repository)
   }
 
   private forcePushWithLease = () => {
+    this.closeDropdown()
     this.props.dispatcher.confirmOrForcePush(this.props.repository)
   }
 
   private pull = () => {
+    this.closeDropdown()
     this.props.dispatcher.pull(this.props.repository)
   }
 
   private fetch = () => {
+    this.closeDropdown()
     this.props.dispatcher.fetch(
       this.props.repository,
       FetchType.UserInitiatedTask
     )
   }
 
+  private getDropdownContentRenderer(
+    itemTypes: ReadonlyArray<DropdownItemType>
+  ) {
+    return () => {
+      return (
+        <>
+          {/*
+          This <div> is not interactive, but its children are, so we need to
+          disable the jsx-a11y/no-static-element-interactions rule.
+          */}
+          {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
+          <div
+            className="push-pull-dropdown"
+            onKeyDown={this.onDropdownKeyDown}
+          >
+            {itemTypes.map(this.renderDropdownItem)}
+          </div>
+        </>
+      )
+    }
+  }
+
+  private onDropdownKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    // Allow using Up and Down arrow keys to navigate the dropdown items
+    // (equivalent to Tab and Shift+Tab)
+    if (event.key !== 'ArrowDown' && event.key !== 'ArrowUp') {
+      return
+    }
+
+    event.preventDefault()
+    const dropdown = event.currentTarget
+    const items = dropdown.querySelectorAll<HTMLElement>(
+      `.${DropdownItemClassName}`
+    )
+    const focusedItem = dropdown.querySelector<HTMLElement>(':focus')
+    if (!focusedItem) {
+      return
+    }
+
+    const focusedIndex = Array.from(items).indexOf(focusedItem)
+    const nextIndex =
+      event.key === 'ArrowDown' ? focusedIndex + 1 : focusedIndex - 1
+    // http://javascript.about.com/od/problemsolving/a/modulobug.htm
+    const nextItem = items[(nextIndex + items.length) % items.length]
+    nextItem?.focus()
+  }
+
   public render() {
+    return this.renderButton()
+  }
+
+  private getDropdownItemWithType(type: DropdownItemType): DropdownItem {
+    const { remoteName } = this.props
+
+    switch (type) {
+      case DropdownItemType.Fetch:
+        return {
+          title: `Fetch ${remoteName}`,
+          description: `Fetch the latest changes from ${remoteName}`,
+          action: this.fetch,
+          icon: syncClockwise,
+        }
+      case DropdownItemType.ForcePush:
+        return {
+          title: `Force push ${remoteName}`,
+          description: (
+            <>
+              Overwrite any changes on {remoteName} with your local changes
+              <br />
+              <br />
+              <div className="warning">
+                <span className="warning-title">Warning:</span> A force push
```

Comment:
Given the user might have disabled the confirmation popup, and this warning doesn't add more steps, I don't feel strongly about removing the warning entirely. But I do agree we should look for consistency in those warnings 😅 I will use one of those styles (yellow octicon + smaller font with default font color) and see if it looks nice.

Reply:
After a few attempts, I'm unable to find a combination for the warning hint that looks well in that button, other than what was mocked and implemented 😕 

So if you don't mind, I'm gonna leave it as it is for now, feature flag this, and we can discuss it at some other point (probably along with the rest of examples you presented in your comment 🙇‍♂️ )

Reply:
> Given the user might have disabled the confirmation popup, and this warning doesn't add more steps, I don't feel strongly about removing the warning entirely

I don't feel super strongly on this. But to clarify cause I can see my wording wasn't great,  I did not mean to remove it entirely. Only remote it when the user is also going to get the confirmation popup. So that is not like, I read this warning, ok I am going to do it anyway, and then I get asked again. :P

Reply:
Ohhh ok, my bad I didn't get that 😳 That sounds very reasonable to me, I can tackle that in a different PR. Thanks!